### PR TITLE
rxm: Avoid accessing base domain name if it is not set.

### PR DESCRIPTION
**Pull request by bturrubiates**
@a-ilango @shefty @goodell 

Signed-off-by: Ben Turrubiates <bturrubi@cisco.com>

### CODE REVIEW

**Comment by lanl-ompi** [comment]
bot:retest

File: prov/rxm/src/rxm_init.c [hunk 1]
```diff
@@ -109,13 +109,23 @@ int rxm_alter_base_info(struct fi_info *base_info, struct fi_info **layer_info)
 	*info->ep_attr = *rxm_info.ep_attr;
 	*info->domain_attr = *rxm_info.domain_attr;
 
-	if (!(info->domain_attr->name =
-				ofi_strdup_add_prefix(base_info->domain_attr->name,
-					rxm_info.domain_attr->name))) {
```
Comment (a-ilango) [inline prov/rxm/src/rxm_init.c]: The assumption for the original code is the underlying provider is expected to fill in base_info->domain_attr->name. That's why the check for that was skipped. @shefty Is this a reasonable assumption to make?

**Comment by shefty** [comment]
Where are we at in the code flow here?  Is this the output from an fi_getinfo call?

**Comment by bturrubiates** [comment]
> Where are we at in the code flow here? Is this the output from an fi_getinfo call?

This gets called as part of `rxm_getinfo`.

**Comment by a-ilango** [comment]
At this point we have an fi_info structure from the base provider (thru fi_getinfo) which we are modifying before sending it to the caller of rxm_getinfo.

**Comment by shefty** [comment]
But is base_info the output from fi_getinfo?  Was the change in response to a travis warning?

**Comment by bturrubiates** [comment]
> But is base_info the output from fi_getinfo? Was the change in response to a travis warning?

`base_info` is the output of `fi_getinfo` for the provider that `rxm` will layer over. This change was in response to our internal Jenkins seg faulting on `fi_info`. The usNIC provider currently doesn't set the `domain_attr->name` field. I'll be submitting a fix to set it, but for now I thought it should handle this more gracefully. This fix makes the domain attribute `name` field be `rxm` if the underlying base provider doesn't populate the `name` field of the domain attr.

I could change it to behave some other way, this is just the first thing that came to mind and `rxm` was already in the layered domain attribute.

**Comment by shefty** [comment]
I don't object to this code, but I'd hate to have to repeat it in every layered provider.  There may be several -- at least 3 are under development that I'm aware of.  We need to find some way of pushing this into a common area.

**Comment by a-ilango** [comment]
This can be handled in ofi_strdup_add_prefix() instead of here. That function can return "<prefix>_" if "name" arg is NULL. The underscore is needed since ofi_strdup_less_prefix() expects it.

**Comment by shefty** [comment]
If prefix=XXX, does ofi_strdup_add_prefix() prepend XXX to the string or XXX_?

**Comment by a-ilango** [comment]
ofi_strdup_add_prefix() prepends XXX_

**Comment by bturrubiates** [comment]
> This can be handled in ofi_strdup_add_prefix() instead of here. That function can return "_" if "name" arg is NULL. The underscore is needed since ofi_strdup_less_prefix() expects it.

If `name` is `NULL` and `prefix` is `rxm`, then is the expected output `rxm_`?

**Comment by a-ilango** [comment]
> If name is NULL and prefix is rxm, then is the expected output rxm_?

Yes. That's what I meant. I had specified it as "\<prefix\>\_" but somehow it came out just as "\_". Probably a markdown quirk. I had to use escapes in this comment.

**Comment by bturrubiates** [comment]
> Yes. That's what I meant. I had specified it as "<prefix>_" but somehow it came out just as "_". Probably a markdown quirk. I had to use escapes in this comment.

Okay, I can revise my PR.

**Comment by bturrubiates** [comment]
Updated.

**Comment by a-ilango** [comment]
Thanks! Looks good.

**Pull request merged**